cmd/samples/recipes/cancelactivity/workflow.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/cadence/activity"
+	"go.uber.org/cadence/workflow"
+	"go.uber.org/zap"
+)
+
+/**
+ * This is the cancel activity workflow sample.
+ */
+
+// ApplicationName is the task list for this sample
+const ApplicationName = "cancelGroup"
+
+// This is registration process where you register all your workflows
+// and activity function handlers.
+func init() {
+	workflow.Register(Workflow)
+	activity.Register(activityToBeCanceled)
+	activity.Register(activityToBeSkipped)
+	activity.Register(cleanupActivity)
+}
+
+// Workflow workflow decider
+func Workflow(ctx workflow.Context) error {
+	ao := workflow.ActivityOptions{
+		ScheduleToStartTimeout: time.Minute,
+		StartToCloseTimeout:    time.Minute * 30,
+		HeartbeatTimeout:       time.Second * 5,
+		WaitForCancellation:    true,
+	}
+	ctx = workflow.WithActivityOptions(ctx, ao)
+	logger := workflow.GetLogger(ctx)
+	logger.Info("cancel workflow started")
+
+	defer func() {
+		newCtx, _ := workflow.NewDisconnectedContext(ctx)
+		err := workflow.ExecuteActivity(newCtx, cleanupActivity).Get(ctx, nil)
+		if err != nil {
+			logger.Error("Cleanup activity failed", zap.Error(err))
+		}
+	}()
+
+	var result string
+	err := workflow.ExecuteActivity(ctx, activityToBeCanceled).Get(ctx, &result)
+	logger.Info(fmt.Sprintf("activityToBeCanceled returns %v, %v", result, err))
+
+	_ = workflow.ExecuteActivity(ctx, activityToBeSkipped).Get(ctx, nil)
```

Comment:
I wouldn't ignore error here and print its value as well

Reply:
done